lib/data/external/AzureClient.js

```diff
@@ -22,12 +23,28 @@ class AzureClient {
             this._azureStorageEndpoint);
         this._dataStoreName = config.dataStoreName;
         this._bucketMatch = config.bucketMatch;
-        if (config.proxy) {
-            const parsedUrl = url.parse(config.proxy);
+        if (config.proxy && config.proxy.url) {
```

Comment:
Ideally we could have a few helper methods or classes to do this prep job in an homogeneous manner for backends that support a proxy. I'm good with this though considering the time constraints.